### PR TITLE
feat: public projects

### PR DESCRIPTION
**Pull request by Sparkier**
Public project can now be viewed even if not logged in.

I will update the UI to only show what's applicable to the user in a future PR.

### CODE REVIEW

**Comment by Sparkier** [comment]
image

File: backend/zeno_backend/database/select.py [hunk 1]
```diff
@@ -152,6 +152,49 @@ def projects(user: User) -> list[Project]:
         db.disconnect()
 
 
+def public_projects() -> list[Project]:
+    """Fetch all publically accessible projects.
+
+    Returns:
+        list[Project]: all publically accessible projects.
```
Comment (cabreraalex) [inline backend/zeno_backend/database/select.py:159]: nit: publicly

File: backend/zeno_backend/util.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    """Check whether a project is public.
+
+    Args:
+        project_uuid (str): uuid of the project to be ckecked.
+
+    Returns:
+        bool: whether the project is public.
+    """
+    db = Database()
+    public = db.connect_execute_return(
```
Comment (cabreraalex) [inline backend/zeno_backend/util.py:22]: Should this be in the select file of database folder?

**Review by cabreraalex — APPROVED** [review]

**Comment by cabreraalex** [comment]
Fixes https://linear.app/zenoml/issue/ZEN-57/backend-project-changes-only-for-authorized-users I believe

**Pull request merged**